task/build-paketo-ubi-builder/0.1/build-paketo-ubi-builder.yaml

```diff
@@ -0,0 +1,527 @@
+      default: "false"
+      type: string
+    - name: IMAGE
+      description: Reference of the image that pack will produce.
+      type: string
+    - name: PLATFORM
+      description: The VM OS type to be used to run the podman container doing the build
+      default: "linux-mlarge/amd64"
+      type: string
+    - name: SOURCE_ARTIFACT
+      description: The Trusted Artifact URI pointing to the artifact with
+        the application source code.
+      type: string
+    - name: SOURCE_CODE_DIR
+      description: The directory containing the code source
+      default: .
+      type: string
+    - name: STORAGE_DRIVER
+      description: Storage driver to configure for buildah
+      default: vfs
+      type: string
+    - name: TLSVERIFY
+      description: Verify the TLS on the registry endpoint (for push/pull to a non-TLS registry)
+      type: string
+      default: "true"
+    - name: caTrustConfigMapKey
+      description: The name of the key in the ConfigMap that contains the CA bundle
+        data.
+      type: string
+      default: ca-bundle.crt
+    - name: caTrustConfigMapName
+      description: The name of the ConfigMap to read CA bundle data from.
+      type: string
+      default: trusted-ca
+  results:
+    - name: "IMAGE_URL"
+      description: "Image repository and tag where the built image was pushed"
+      type: string
+    - name: "IMAGE_DIGEST"
+      description: "Digest of the image just built"
+      type: string
+    - name: "IMAGE_REF"
+      description: "Image reference of the built image"
+      type: string
+    - name: "BASE_IMAGES_DIGESTS"
+      description: "Digests of the base images used for build"
+      type: string
+    - name: "SBOM_BLOB_URL"
+      description: "SBOM Image URL"
+      type: string
+  volumes:
+    - name: ssh
+      secret:
+        secretName: "multi-platform-ssh-$(context.taskRun.name)"
+    - name: shared
+      emptyDir: {}
+    - name: varlibcontainers
+      emptyDir: {}
+    - configMap:
+        items:
+          - key: $(params.caTrustConfigMapKey)
+            path: ca-bundle.crt
+        name: $(params.caTrustConfigMapName)
+        optional: true
+      name: trusted-ca
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    env:
+      - name: BUILDER_IMAGE
+        value: $(params.BUILDER_IMAGE)
+      - name: CONTEXT
+        value: $(params.CONTEXT)
+      - name: HERMETIC
+        value: $(params.HERMETIC)
+      - name: IMAGE
+        value: $(params.IMAGE)
+      - name: PLATFORM
+        value: $(params.PLATFORM)
+      - name: SOURCE_CODE_DIR
+        value: $(params.SOURCE_CODE_DIR)
+      - name: STORAGE_DRIVER
+        value: $(params.STORAGE_DRIVER)
+      - name: TLSVERIFY
+        value: $(params.TLSVERIFY)
+
+    volumeMounts:
+      - mountPath: /shared
+        name: shared
+      - mountPath: /var/workdir
+        name: workdir
+  steps:
+      - name: use-trusted-artifact
```

Comment:
This task uses trusted artifacts, so task name should have suffix `-oci-ta` (build-paketo-ubi-builder-oci-ta)

Also, should this support non trusted-artifacts workflows? If yes, then it should have two versions, standard docker build version and then recipe.yaml to generate TA version.

Reply:
> This task uses trusted artifacts, so task name should have suffix `-oci-ta` (build-paketo-ubi-builder-oci-ta)

Ok. I will then rename it

Reply:
Task renamed. See: 92f55eac

Reply:
> Also, should this support non trusted-artifacts workflows?

I will ask the question to the product manager - Naina Singh and will keep you informed. That could then addressed maybe in a separate PR.

Reply:
we can stay with TA task only, as in future we should have only TA tasks, unless you need to use PVCs